### ISSUE TITLE
Importing Keras should not execute code, makes pydoc sad

### ISSUE THREAD

**Issue by pgunn** [keras-team/keras#8311]
Right now, importing Keras executes code (the backend component is one example). This is surprising behaviour, and makes anything else that imports Keras slow and a little noisy (it tells you what frontend it would use). This happens even if somebody does pydoc on a library that happens to import Keras. In an ideal world, running pydoc on Keras or anything importing it would not run any initialisation code embedded in the libraries.

(I can try to put together a PR to fix this if people are open to the fix)

**Comment by datumbox** [comment]
Confirmed. As pointed out this is related to https://github.com/fchollet/keras/issues/8353. 

Here is Keras 2.0.8:
```
$ python

Python 2.7.6 (default, Oct 26 2016, 20:30:19) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import keras
Using TensorFlow backend.
>>> keras.__version__
'2.0.8'
>>> 
```

Here is Keras 2.0.9:
```
$ python

Python 2.7.6 (default, Oct 26 2016, 20:30:19) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import keras
Using TensorFlow backend.
2017-11-03 09:44:12.218649: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2017-11-03 09:44:12.440512: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties: 
name: Quadro K2200 major: 5 minor: 0 memoryClockRate(GHz): 1.124
pciBusID: 0000:03:00.0
totalMemory: 3.95GiB freeMemory: 3.42GiB
2017-11-03 09:44:12.617718: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 1 with properties: 
name: Quadro K2200 major: 5 minor: 0 memoryClockRate(GHz): 1.124
pciBusID: 0000:81:00.0
totalMemory: 3.95GiB freeMemory: 3.91GiB
2017-11-03 09:44:12.617861: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Device peer to peer matrix
2017-11-03 09:44:12.617927: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1051] DMA: 0 1 
2017-11-03 09:44:12.617950: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 0:   Y N 
2017-11-03 09:44:12.617966: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1061] 1:   N Y 
2017-11-03 09:44:12.618000: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Quadro K2200, pci bus id: 0000:03:00.0, compute capability: 5.0)
2017-11-03 09:44:12.618028: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:1) -> (device: 1, name: Quadro K2200, pci bus id: 0000:81:00.0, compute capability: 5.0)
>>> keras.__version__
'2.0.9'
>>> 
```

**Comment by datumbox** [comment]
This is patched with #8377
